Resources/public/js/views/services/ez-discoverybarviewservice.js

```diff
@@ -20,10 +20,62 @@ YUI.add('ez-discoverybarviewservice', function (Y) {
      * @extends eZ.ViewService
      */
     Y.eZ.DiscoveryBarViewService = Y.Base.create('discoveryBarViewService', Y.eZ.ViewService, [Y.eZ.SideViewService], {
-
         initializer: function () {
             this.on('*:viewTrashAction', this._redirectToTrashView);
             this.on('*:viewSearchAction', this._redirectToSearchView);
+            this.on('*:browseAction', this._fireContentDiscover);
+        },
+
+        /**
+         * Checks whether the currently active view is a LocationViewView.
+         *
+         * @method _isLocationViewDisplayed
+         * @protected
+         * @return {Boolean}
+         */
+        _isLocationViewDisplayed: function () {
+            return this.get('app').get('activeView') instanceof Y.eZ.LocationViewView;
+        },
+
+        /**
+         * Launch UDW to the current location view if there is one.
+         *
+         * @method _fireContentDiscover
+         * @protected
+         */
+        _fireContentDiscover: function () {
+            var startingLocationId,
+                rootDepth;
+
+            if (this._isLocationViewDisplayed()) {
+               startingLocationId = this.get('app').get('activeView').get('location').get('id');
+               rootDepth = 1;
+            }
+
+            this.fire('contentDiscover', {
+                config: {
+                    title: Y.eZ.trans('universaldiscovery.content.browser', {}, 'universaldiscovery'),
```

Comment:
`universaldiscovery` is not the right context for this string, `bar` would be more suited.